test/unit/spec/nbextensions/bulkImportCell/main-spec.js

```diff
@@ -19,82 +19,101 @@ define([
             Jupyter.narrative = null;
         });
 
-        beforeEach(() => {
-            // mock the notebook for the main module
-            const cell = Mocks.buildMockCell('code', 'app-bulk-import');
-            // the mock has two cells, a plain mock code cell, and
-            // a mock BulkImportCell
-            Jupyter.notebook = Mocks.buildMockNotebook({
-                cells: [cell],
-                fullyLoaded: true,
+        describe('on start', () => {
+            beforeEach(() => {
+                // create a mock bulkImportCell and add it to the mock notebook
+                const cell = Mocks.buildMockCell('code', 'app-bulk-import');
+                // mock the notebook for the main module
+                Jupyter.notebook = Mocks.buildMockNotebook({
+                    cells: [cell],
+                    fullyLoaded: true,
+                });
             });
-        });
 
-        afterEach(() => {
-            Jupyter.notebook = null;
-        });
+            afterEach(() => {
+                Jupyter.notebook = null;
+            });
 
-        it('should have a load_ipython_extension function', () => {
-            expect(Main.load_ipython_extension).toBeDefined();
-        });
+            it('should have a load_ipython_extension function', () => {
+                expect(Main.load_ipython_extension).toBeDefined();
+            });
 
-        it('should initialize the notebook on start', () => {
-            // expect that setupNotebook was called, which we can
-            // proxy by checking on Jupyter.notebook.get_cells.
-            spyOn(Jupyter.notebook, 'get_cells').and.callThrough();
-            return Main.load_ipython_extension().then(() => {
-                expect(Jupyter.notebook.get_cells).toHaveBeenCalled();
+            it('should initialize the notebook', () => {
+                // expect that setupNotebook was called, which we can
+                // proxy by checking on Jupyter.notebook.get_cells.
+                spyOn(Jupyter.notebook, 'get_cells').and.callThrough();
+                spyOn(BulkImportCell, 'make').and.callFake(() => {
+                    // fake out the call to 'make' the bulk import cell
+                    // as the mock does not have all the metadata
```

Comment:
that creates a error dialog, so faking out the function means the error never gets called